common/build/eslint-plugin-fluid/src/test/example/no-unchecked-record-access/fileWithOnlyRecordAccess.ts

```diff
@@ -0,0 +1,119 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+const record = { a: 1, b: 2 };
+const recordA = record.a; // ok: Accessing number property 'a' directly
+const recordB = record.b; // ok: Accessing number property 'b' directly
+
+interface NestedIndexProps {
+	nested: {
+		[key: string]: string;
+	};
+}
+type StaticType = { a: string; b: string };
+type IndexSignatureType = { [key: string]: string };
+type NullableIndexSignatureType = { [key: string]: string | null };
+type UndefinableIndexSignatureType = { [key: string]: string | undefined };
+const nullableIndexedRecord: NullableIndexSignatureType = { a: "hello", b: null };
+nullableIndexedRecord.a; // ok: Accessing index property 'a' without requiring a particular result
+nullableIndexedRecord.a.length; // defect: Accessing length of index property 'a', but 'a' might be undefined or null
+nullableIndexedRecord.b?.length; // ok: Using optional chaining to access length safely handles 'undefined'
+const undefinableIndexedRecord: UndefinableIndexSignatureType = { a: "hello", b: undefined };
+undefinableIndexedRecord.a; // ok: Accessing index property 'a' without requiring a particular result
+undefinableIndexedRecord.a.length; // defect: Accessing length of index property 'a', but 'a' might be undefined or null
+undefinableIndexedRecord.b?.length; // ok: Using optional chaining to access length safely handles 'undefined'
```

Comment:
Let's organize these better. Before adding these there was the one-off nested, then statics, then regular index. These feel like special cases of index and make sense to have after those.